File: LeetCode/LeetCode Practice Problems/1817- Finding the Users Active Minutes.cpp
class Solution {
public:
    vector<int> findingUsersActiveMinutes(vector<vector<int>>& logs, int k) {
        map<int,set<int>> m;
        vector<int> ans(k,0);
        
        for(auto i: logs) {
            m[i[0]].insert(i[1]);
        }
        
        
        for(auto i:m) {
            ans[i.second.size() - 1]++;
        }
        
        return ans;
        
    }
};

File: LeetCode/LeetCode Practice Problems/1817-Finding the Users Active Minutes.java
class Solution {
    public int[] findingUsersActiveMinutes(int[][] logs, int k) {
        HashMap<Integer,Set<Integer>> map = new HashMap();
        
        for(int log[]: logs) {
            Set<Integer> s = map.getOrDefault(log[0], new HashSet());
            s.add(log[1]);
            map.put(log[0] , s);
        }
        
        int ans[] = new int[k];
        
        for(int key: map.keySet()) {
            int value = map.get(key).size() - 1;
            if(value < k) {
                ans[value]++;
            }
        }
        
        return ans;
    }
}
